inception-ui-kb/src/main/java/de/tudarmstadt/ukp/inception/ui/kb/feature/SubjectObjectFeatureEditor.java

```diff
@@ -371,64 +370,136 @@ private KBHandle getSelectedKBItem()
+                }
+                else {
+                    if (traits.getScope() != null) {
+                        handles.addAll(
+                                kbService
+                                        .listInstancesForChildConcepts(kb, traits.getScope(), false,
+                                                50)
+                                .stream().filter(inst -> inst.getUiLabel().contains(aTypedString))
+                                .collect(Collectors.toList()));
                     }
                     else {
-                        if (traits.getScope() != null) {
-                            handles.addAll(kbService.listInstances(kb, traits.getScope(), false)
-                                .stream()
-                                .filter(inst -> inst.getUiLabel().contains(aTypedString))
-                                .collect(Collectors.toList()));
-                        }
-                        else {
-                            for (KBHandle concept : kbService.listConcepts(kb, false)) {
-                                handles.addAll(
-                                    kbService.listInstances(kb, concept.getIdentifier(), false));
-                            }
+                        for (KBHandle concept : kbService.listConcepts(kb, false)) {
+                            handles.addAll(
+                                kbService.listInstances(kb, concept.getIdentifier(), false));
                         }
                     }
                 }
             }
         }
-        catch (Exception e) {
-            // LOG.error("Unable to read traits", e);
-            error("Unable to read traits: " + ExceptionUtils.getRootCauseMessage(e));
-            IPartialPageRequestHandler target = RequestCycle.get()
-                .find(IPartialPageRequestHandler.class);
-            if (target != null) {
-                target.addChildren(getPage(), IFeedback.class);
+        return handles;
+    }
+
+    private List<KBHandle> getConcepts(ConceptFeatureTraits traits, Project project,
+        AnnotationActionHandler aHandler, String aTypedString)
+    {
+        List<KBHandle> handles = new ArrayList<>();
+        if (traits.getRepositoryId() != null) {
+            // If a specific KB is selected, get its instances
+            Optional<KnowledgeBase> kb = kbService
+                .getKnowledgeBaseById(project, traits.getRepositoryId());
+            if (kb.isPresent()) {
+                //TODO: (#122) see ConceptFeatureEditor
+                if (kb.get().isSupportConceptLinking()) {
+                    handles.addAll(
+                        listLinkingInstances(kb.get(), () -> getEditorCas(aHandler), aTypedString));
+                }
+                else {
+                    if (traits.getScope() != null) {
+                        handles = kbService.listChildConcepts(kb.get(), traits.getScope(),
+                            false)
```

Comment:
Badly formatted. I suggest you do a single CTRL-F for the correctly formatting the whole file and then commit it alone without any other functional change, in order not to mix important changes with cosmetic changes.